artiq/applets/progress_bar.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+
+from PyQt5 import QtWidgets
+
+from artiq.applets.simple import SimpleApplet
+
+
+class ProgressWidget(QtWidgets.QProgressBar):
+    def __init__(self, args):
+        QtWidgets.QProgressBar.__init__(self)
+        self.setMaximum(args.max)
+        self.setMinimum(args.min)
+        self.dataset_n = args.n
+        self.dataset_n_max = args.n_max
+
+    def data_changed(self, data, mods):
+        try:
+            n = int(round(data[self.dataset_n][1]))
+            n_max = int(round(data[self.dataset_n_max][1]))
+        except (KeyError, ValueError, TypeError):
+            n = 0
+            n_max = 0
+        self.setMaximum(n_max)
+        self.setValue(n)
+
+
+def main():
+    applet = SimpleApplet(ProgressWidget)
+    applet.add_dataset("n", "counter")
```

Comment:
Call it "value" to match Qt terminology.

Reply:
done